rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -200,8 +200,17 @@ namespace ros {
         while( true )
         {
           int data = hardware_.read();
-          if( data < 0 )
+          if( data < 0 ) {
+#if defined(USE_WIFICON)
+		    if (!hardware_.isConnectionUp()) {		// check connection, if it is broken try to repair it N.B. should really add interrupt timer as this blocks !
+				configured_ = false;
+				return -3;
+			} else
+				break;
+#else
```

Comment:
I'm not keen on having to add this housekeeping logic to rosserial_client. Is there a way to manage the wifi connection without requiring this modification?